docs/guides/latest/volumesnapshot/statefulset.md

```diff
@@ -0,0 +1,820 @@
+```
+
+Let's find out the actual snapshot name that will be saved in the Google Cloud by the following command,
+
+```console
+kubectl get volumesnapshot source-data-stash-demo-0-1563177551 -n demo -o yaml
+```
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1alpha1
+kind: VolumeSnapshot
+metadata:
+  creationTimestamp: "2019-07-15T07:59:13Z"
+  finalizers:
+  - snapshot.storage.kubernetes.io/volumesnapshot-protection
+  generation: 4
+  name: source-data-stash-demo-0-1563177551
+  namespace: demo
+  resourceVersion: "18764"
+  selfLink: /apis/snapshot.storage.k8s.io/v1alpha1/namespaces/demo/volumesnapshots/source-data-stash-demo-0-1563177551
+  uid: 6f3b49a9-a6d6-11e9-9f3a-42010a800050
+spec:
+  snapshotClassName: default-snapshot-class
+  snapshotContentName: snapcontent-6f3b49a9-a6d6-11e9-9f3a-42010a800050
+  source:
+    apiGroup: null
+    kind: PersistentVolumeClaim
+    name: source-data-stash-demo-0
+status:
+  creationTime: "2019-07-15T07:59:14Z"
+  readyToUse: true
+  restoreSize: 1Gi
+```
+
+Here, `spec.snapshotContentName` field specifies the name of the `VolumeSnapshotContent` crd. It also represents the actual snapshot name that has been saved in Google Cloud. If we navigate to the `Snapshots` tab in the GCP console, we should see snapshot `snapcontent-6f3b49a9-a6d6-11e9-9f3a-42010a800050` has been stored successfully.
+
+<figure align="center">
+  <img alt="Stash Backup Flow" src="/docs/images/guides/latest/volumesnapshot/statefulset.png">
+<figcaption align="center">Fig: Snapshots in GCE Bucket</figcaption>
+</figure>
+
+## Restore PVC from VolumeSnapshot
+
+This section will show you how to restore PVCs from the snapshots we have taken in earlier section.
+
+**Create RestoreSession :**
+
+At first, we have to create a `RestoreSession` crd to restore PVCs from respective the snapshots.
+
+Below is the YAML of the `RestoreSesion` crd that we are going to create,
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: RestoreSession
+metadata:
+  name: restore-pvc
+  namespace: demo
+spec:
+  driver: VolumeSnapshotter
+  target:
+    replicas : 3
+    volumeClaimTemplates:
+      - metadata:
+          name: restore-data-restore-demo-${POD_ORDINAL}
+        spec:
+          accessModes: [ "ReadWriteOnce" ]
+          storageClassName: "standard"
+          resources:
+            requests:
+              storage: 1Gi
+          dataSource:
+            kind: VolumeSnapshot
+            name: source-data-stash-demo-${POD_ORDINAL}-1563177551
+#           name: source-data-stash-demo-0-1563177551
+            apiGroup: snapshot.storage.k8s.io
+```
+
+Here,
+
+- `spec.target.replicas`: `spec.target.replicas` specify the number of replicas of a StatefulSet whose volumes was backed up and Stash uses this field to dynamically create the desired number of PVCs and initialize them from respective or Specific VolumeSnapShots.
+- `spec.target.volumeClaimTemplates`:
+  - `metadata.name` is the name of the restored `PVC`.
```

Comment:
- `metadata.name` is a template for the name of the restored PVC that will be created by Stash. You have to provide this name template to match with your desired StatefulSet's PVC. For example, if you want deploy a StatefulSet named `stash-demo` with `volumeClaimTemplate` name `my-volume`, your StatefulSet's PVC will be `my-volume-stash-demo-0`, `my-volume-stash-demo-1` and so on. In this case, you have to provide `volumeClaimTemplate` name in `RestoreSession` in the following format:
	```console
	<volume claim name>-<statefulset name>-${POD_ORDINAL}
	```
	So for the above example, `volumeClaimTemplate` name for `RestoreSession` will be `my-volume-stash-demo-${POD_ORDINAL}`.